### PR TITLE
MODCON-49 implemented get publication details

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
* https://issues.folio.org/browse/MODCON-49

## Approach
* implemented get publication details

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/swagger.api/schemas/publication_schema.yaml [hunk 1]
```diff
@@ -25,13 +25,47 @@ PublicationResponse:
...
+  title: Publication details response
+  properties:
+    id:
+      description: id of publication record
+      $ref: "common.yaml#/uuid"
+    status:
+      type: string
+      $ref: "publication_schema.yaml#/PublicationStatus"
+    dateTime:
+      description: id of publication record
```
Comment (SerhiiNosko) [inline src/main/resources/swagger.api/schemas/publication_schema.yaml:46]: description should be updated for dateTime and request fields

File: src/main/java/org/folio/consortia/repository/PublicationTenantRequestRepository.java [hunk 2]
```diff
@@ -3,9 +3,14 @@
...
 import org.folio.consortia.domain.entity.PublicationTenantRequestEntity;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface PublicationTenantRequestRepository extends JpaRepository<PublicationTenantRequestEntity, UUID> {
+
+  Page<PublicationTenantRequestEntity> findByPcState_Id(UUID publicationId, Pageable pageable);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/repository/PublicationTenantRequestRepository.java:14]: Can it be renamed to findByPcStateId?

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java [hunk 3]
```diff
@@ -69,6 +76,48 @@ public PublicationResponse publishRequest(UUID consortiumId, PublicationRequest
...
+
+  @Override
+  public PublicationDetailsResponse getPublicationDetails(UUID consortiumId, UUID publicationId) {
+    log.debug("getById:: Trying to retrieve publicationDetails by consortiumId: {} and publicationId id: {}", consortiumId, publicationId);
+
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    var publicationStatusEntity = publicationStatusRepository.findById(publicationId)
+      .orElseThrow(() -> new ResourceNotFoundException("publicationId", String.valueOf(publicationId)));
+
+    var publicationTenantRequestEntity = publicationTenantRequestRepository.findByPcState_Id(publicationId, PageRequest.of(0, Integer.MAX_VALUE));
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java:88]: It is list in response, so I propose to rename to ptrEntities

File: src/main/resources/swagger.api/schemas/publication_schema.yaml [hunk 4]
```diff
@@ -25,13 +25,47 @@ PublicationResponse:
...
   additionalProperties: false
 
 PublicationStatus:
   description: publication status
   enum: [ "IN_PROGRESS", "ERROR", "COMPLETE" ]
+
+PublicationStatusError:
+  description: publication status error
+  properties:
+    tenantId:
```
Comment (SerhiiNosko) [inline src/main/resources/swagger.api/schemas/publication_schema.yaml:66]: I propose to add also status code here in response, especially to distinguish 400x and 500x errors, this field already exists in db table

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java [hunk 5]
```diff
@@ -69,6 +76,48 @@ public PublicationResponse publishRequest(UUID consortiumId, PublicationRequest
 
     return buildPublicationResponse(createdPublicationEntity.getId());
   }
+
+  @Override
+  public PublicationDetailsResponse getPublicationDetails(UUID consortiumId, UUID publicationId) {
+    log.debug("getById:: Trying to retrieve publicationDetails by consortiumId: {} and publicationId id: {}", consortiumId, publicationId);
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java:82]: getById->getPublicationDetails

File: src/main/resources/swagger.api/schemas/publication_schema.yaml (hunk 4 above)
Comment (grabsefx) [reply src/main/resources/swagger.api/schemas/publication_schema.yaml:66]: added errorCode

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java (hunk 3 above)
Comment (grabsefx) [reply src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java]: done

File: src/main/java/org/folio/consortia/repository/PublicationTenantRequestRepository.java (hunk 2 above)
Comment (grabsefx) [reply src/main/java/org/folio/consortia/repository/PublicationTenantRequestRepository.java]: yes, renamed

**Review by SerhiiNosko — APPROVED** [review]

**Review by Dmitriy-Butramyou — APPROVED** [review]

**Pull request merged**